src/Servers/HttpSys/test/FunctionalTests/ResponseCachingTests.cs

```diff
@@ -85,6 +85,27 @@ public async Task Caching_WithoutContentType_NotCached()
             }
         }
 
+        [ConditionalFact]
+        public async Task Caching_304_NotCached()
+        {
+            var requestCount = 1;
+            using (Utilities.CreateHttpServer(out string address, httpContext =>
+            {
+                // 304 responses are not themselves cachable. Their cache header mirrors the resource's original cache header.
+                httpContext.Response.StatusCode = StatusCodes.Status304NotModified;
+                httpContext.Response.ContentType = "some/thing"; // Http.Sys requires content-type for caching
+                httpContext.Response.Headers["x-request-count"] = (requestCount++).ToString();
+                httpContext.Response.Headers["Cache-Control"] = "public, max-age=10";
+                httpContext.Response.ContentLength = 10;
+                return httpContext.Response.Body.WriteAsync(new byte[10], 0, 10);
+            }))
+            {
+                address += Guid.NewGuid().ToString(); // Avoid cache collisions for failed tests.
```

Comment:
Why does this test do this, but not the following test?

Reply:
There's no way to clear the cache between tests, this helps avoid collisions. Not sure why only half the tests have them, I'll add them to the other half.